server_rpc.py

```diff
@@ -9,22 +9,35 @@
 @dispatcher.add_method
 def fire_client(**kwargs):
     resp = ""
-    if ("npatient" in kwargs.keys()) and ("serve_ip" in kwargs.keys()) and ("serve_port" in kwargs.keys()) and ("go_client_name" in kwargs.keys()):
+    if (
+            ("npatient" in kwargs.keys()) and
+            ("serve_ip" in kwargs.keys()) and
```

Comment:
This is inefficient. Try this - 
`if ["npatient","serve_ip", "serve_port" ,"go_client_name" ] == list(kwargs.keys())`